lib/driver.js

```diff
@@ -135,11 +135,15 @@ const MEMOIZED_FUNCTIONS = [
 ];
 
 class AndroidUiautomator2Driver extends BaseDriver {
-  constructor (opts = {}, shouldValidateCaps = true) {
+  constructor (opts = {}, shouldValidateCaps = true, driverArgs = {}) {
     // `shell` overwrites adb.shell, so remove
     delete opts.shell;
+    opts.reboot = helpers.parseReboot(driverArgs);
```

Comment:
one universal function like 

```js
function parseKnownArgs (allArgs, knownArgNames) {
  return allArgs.reduce((acc, argName) => {
    if (knownArgNames.includes(argName)) {
      acc[argName] = allArgs[argName];
    }
    return acc;
  }, {});
}
```

should be enough

Reply:
then to call it: super(opts, shouldValidateCaps, Object.assign(opts, parseKnownArgs(driverArgs, ['reboot', 'chromedriverPort'...])));

Reply:
Also, if these are just capabilities then why cannot we simply pass default capabilities map to the appropriate driver?

Reply:
@mykola-mokhnach in many cases they won't be caps. they'll be things the server admin would need to set on the CLI, that we wouldn't want accessible as caps since we don't want end users fiddling with them.

Reply:
in such case it would still make sense to have them in a map rather in separate entries. The only diff would be that they will be applied after user capabilities rather than before

Reply:
i'm confused; the `driverArgs` object is a map of names to values.